azure-pipelines-pr-validation.yml

```diff
@@ -101,15 +101,8 @@ stages:
         useGlobalJson: true
         workingDirectory: '$(Build.SourcesDirectory)'
 
-    # Needed to restore the Microsoft.DevDiv.Optimization.Data.PowerShell package
-    - task: NuGetCommand@2
+    - script: dotnet restore eng\common\internal\Tools.csproj
       displayName: Restore internal tools
-      inputs:
-        command: restore
-        feedsToUse: config
-        restoreSolution: 'eng\common\internal\Tools.csproj'
-        nugetConfigPath: 'NuGet.config'
-        restoreDirectory: '$(Build.SourcesDirectory)\.packages'
```

Comment:
It looks like the `NUGET_PACKAGES` env var changed in our run to point to a different location.  The build script uses that to find this package and so failed.

The fix is to restore via the dotnet cli which will respect the `NUGET_PACKAGES` env var.